src/modules/aws_s3/mappers/bucket_website.ts

```diff
@@ -0,0 +1,112 @@
+import { S3 } from '@aws-sdk/client-s3';
+import { GetBucketWebsiteCommandInput } from '@aws-sdk/client-s3/dist-types/commands/GetBucketWebsiteCommand';
+import { GetBucketWebsiteOutput } from '@aws-sdk/client-s3/dist-types/models/models_0';
+
+import { AwsS3Module } from '..';
+import { AWS, crudBuilder2 } from '../../../services/aws_macros';
+import { Context, Crud2, MapperBase } from '../../interfaces';
+import { Bucket, BucketWebsite } from '../entity';
+
+export class BucketWebsiteMapper extends MapperBase<BucketWebsite> {
+  module: AwsS3Module;
+  entity = BucketWebsite;
+
+  private async getBucketWebsiteForBucket(ctx: Context, bucket: Bucket) {
+    const client = (await ctx.getAwsClient(bucket.region)) as AWS;
+    const input: GetBucketWebsiteCommandInput = {
+      Bucket: bucket.name,
+    };
+    const result = await this.getBucketWebsite(client.s3Client, input);
+    return this.bucketWebsiteMapper(result!, bucket);
+  }
+
+  bucketWebsiteMapper(rawBucketWebsite: GetBucketWebsiteOutput, bucket: Bucket) {
+    const out: BucketWebsite = new BucketWebsite();
+    out.indexDocument = rawBucketWebsite.IndexDocument?.Suffix!;
+    out.errorDocument = rawBucketWebsite.ErrorDocument?.Key;
+    out.bucket = bucket;
+    out.bucketName = bucket.name;
+    return out;
+  }
+
+  equals = (a: BucketWebsite, b: BucketWebsite) =>
+    a.indexDocument === b.indexDocument && a.errorDocument === b.errorDocument;
+
+  getBucketWebsite = crudBuilder2<S3, 'getBucketWebsite'>('getBucketWebsite', input => input);
+
+  cloud = new Crud2<BucketWebsite>({
+    create: async (es: BucketWebsite[], ctx: Context) => {
+      const out = [];
+      for (const e of es) {
+        await this.saveBucketWebsite(ctx, e);
+        out.push(e);
+      }
+      return out;
+    },
+    read: async (ctx: Context, id?: string) => {
+      const out: BucketWebsite[] = [];
+      const buckets =
+        ctx.memo?.cloud?.Bucket && Object.values(ctx.memo?.cloud?.Bucket).length
```

Comment:
Nit. You don't need the `?.` operator in the second half of this because you can only reach it if `ctx.memo.cloud.Bucket` is not null-ish the whole way through.

Could be:

```suggestion
        ctx.memo?.cloud?.Bucket && Object.values(ctx.memo.cloud.Bucket).length
```